scripts/deploy.py

```
@@ -0,0 +1,361 @@
+        try:
+            # Use full path to git executable
+            result = subprocess.run(
+                [str(self.git_exe), '--version'],
+                check=True,
+                capture_output=True,
+                text=True
+            )
+            if not result.stdout.strip().startswith('git version'):
+                raise RuntimeError("Invalid git version output")
+        except subprocess.CalledProcessError as e:
+            raise RuntimeError("Git check failed: %s" % e)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)